errors/index.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const _ = require('lodash');
+
+class ExpectedValueError extends Error {
+    constructor(msg = 'Expected value is not met', ...params) {
+        super(msg, ...params);
+
+        // Maintains proper stack trace for where our error was thrown (only available on V8)
+        if (Error.captureStackTrace) {
+            Error.captureStackTrace(this, ExpectedValueError);
```

Comment:
Should this be on a super class and be extended by the Error classes?

Reply:
Changed this based on the request